zulip_bots/zulip_bots/bots/giphy/giphy.py

```diff
@@ -43,16 +44,17 @@ class GiphyNoResultException(Exception):
     pass
 
 
-def get_url_gif_giphy(keyword, api_key):
+def get_url_gif_giphy(keyword: str, api_key: str) -> Union[int, str]:
     # Return a URL for a Giphy GIF based on keywords given.
     # In case of error, e.g. failure to fetch a GIF URL, it will
     # return a number.
     query = {'s': keyword,
              'api_key': api_key}
     try:
         data = requests.get(GIPHY_TRANSLATE_API, params=query)
-    except requests.exceptions.ConnectionError as e:  # Usually triggered by bad connection.
-        logging.warning(e)
+    except requests.exceptions.ConnectionError as e:
+        # Usually triggered by bad connection.
+        logging.warning('{}'.format(e))
         raise
```

Comment:
Actually, I think it might be better to do this (this just occured to me):
```
logging.exception('Bad connection')
raise
```

See [this documentation](https://docs.python.org/3/library/logging.html#logging.Logger.exception), the exception info should be added to the message, if I'm not wrong.